src/services/emailService.ts

```diff
@@ -0,0 +1,47 @@
+import { none, Option, some } from "fp-ts/lib/Option";
+import * as nodemailer from "nodemailer";
+import { log } from "../utils/logger";
+
+export interface ITransporterOptions {
+  auth: {
+    pass: string;
+    user: string;
+  };
+  host: string;
+  port: number;
+  secure: boolean;
+}
+
+interface ITransporter {
+  // tslint:disable-next-line:no-any
+  sendMail: (mailOptions: IMailOptions) => Promise<any>;
+  verify: () => Promise<true>;
+}
+
+export interface IMailOptions {
+  // tslint:disable-next-line:readonly-array
+  attachments?: Array<{ path: string; cid?: string }>;
+  from: string;
+  html: string;
+  subject: string;
+  text: string;
+  to: string;
+}
+
+export default class EmailService {
+  private transporter: ITransporter;
+  public constructor(transporterConfig: ITransporterOptions) {
+    this.transporter = nodemailer.createTransport(transporterConfig);
+    this.transporter
```

Comment:
I'd put this control (verify) at the top level in app.ts. here it's not clear what is supposed the application to do when the email service construction fails; log the error and go on anyway ?